parser.go

```diff
@@ -49,9 +49,16 @@ func (p *Parser) Trigger() []byte {
 //
 // This will treat "My page" as the target and "click here" as the label for
 // the link.
+//
+// A wikilink can also be embedded, i.e. starts with a bang (!).
+// A few examples of embedded wikilink:
+//
+//  ![[My page]] (simple)
+//  ![[My page#fragment|click here]]
 func (p *Parser) Parse(_ ast.Node, block text.Reader, _ parser.Context) ast.Node {
 	line, seg := block.PeekLine()
-	if !bytes.HasPrefix(line, _open) {
+	start := bytes.Index(line, _open)
+	if start > 1 || (start == 1 && line[0] != '!') {
```

Comment:
I think we can make this clearer with something like:

```
line, seg := ..
var embed bool
switch {
case bytes.HasPrefix(line, _open):
  // not an embed
case bytes.HasPrefix(line, _embedOpen):
  embed = true
default:
  return nil
}
```

Where `_embedOpen = []byte("![[")`.
It makes what we're searching for slightly more obvious without offset munging.